pkg/devfile/parser/data/common/types.go

```diff
@@ -275,6 +275,9 @@ type DevfileParent struct {
 
 	// Bindings of commands to events. Each command is referred-to by its name.
 	Events DevfileEvents `json:"events,omitempty" yaml:"events,omitempty"`
+
+	// StarterProjects is a project that can be used as a starting point when bootstrapping new projects
+	StarterProjects []DevfileStarterProject `json:"starterProjects,omitempty" yaml:"starterProjects,omitempty"`
```

Comment:
Curious, why do we add the `StarterProjects` type to the common package in the parser, rather than under `parser/data/2.0.0`?

Reply:
every devfile struct(1.0, 2.0) needs to satisfy this interface https://github.com/openshift/odo/blob/master/pkg/devfile/parser/data/interface.go#L8 
the methods here used everywhere in adapters code, to make the adapters code work on a common type regardless of devfile version passed, it has been done in this way.